qiskit_experiments/library/characterization/resonator_spectroscopy.py

```diff
@@ -0,0 +1,176 @@
+        circuit
+
+        .. parsed-literal::
+
+                 ┌─┐
+              q: ┤M├
+                 └╥┘
+            c: 1/═╩═
+                  0
+
+        where a spectroscopy pulse is attached to the measurement instruction. When doing
+        readout resonator spectroscopy, each measured IQ point has a frequency dependent
+        phase. Close to the resonance, the IQ points start rotating around in the IQ plan.
+        To create a meaningful signal this experiment therefore uses a custom data processor
+        where the dimensionality reducing SVD is replaced by the absolute value of the IQ
+        point.
+    """
+
+    def __init__(
+        self,
+        qubit: int,
+        frequencies: Iterable[float],
+        backend: Optional[Backend] = None,
+        absolute: bool = True,
+    ):
+        """
+        A spectroscopy experiment run by setting the frequency of the readout drive.
+        The parameters of the GaussianSquare spectroscopy pulse can be specified at run-time.
+        The spectroscopy pulse has the following parameters:
+        - amp: The amplitude of the pulse must be between 0 and 1, the default is 0.1.
+        - duration: The duration of the spectroscopy pulse in samples, the default is 1000 samples.
+        - sigma: The standard deviation of the pulse, the default is duration / 4.
+        - width: The width of the flat-top in the pulse, the default is 0, i.e. a Gaussian.
+
+        Args:
+            qubit: The qubit on which to run readout spectroscopy.
+            frequencies: The frequencies to scan in the experiment, in Hz.
+            backend: Optional, the backend to run the experiment on.
+            absolute: Boolean to specify if the frequencies are absolute or relative to the
+                resonator frequency in the backend.
+        """
+        super().__init__(qubit, frequencies, backend, absolute)
+        self.analysis = ResonatorSpectroscopyAnalysis()
+        self._set_analysis_data_processor()
+
+    def _set_analysis_data_processor(self):
+        """Keep the data processor consistent with the run options."""
+        processor = get_processor(
+            self.run_options.meas_level,
+            self.run_options.meas_return,
+            self.analysis.options.normalization,
+            dimensionality_reduction=ProjectorType.ABS,
+        )
+        self.analysis.set_options(data_processor=processor)
+
+    def set_run_options(self, **fields):
+        """Wrap set run options to keep the data processor consistent."""
+        super().set_run_options(**fields)
+        self._set_analysis_data_processor()
+
+    @property
+    def center_frequency(self) -> float:
+        """Returns the center frequency of the experiment.
+
+        Returns:
+            The center frequency of the experiment.
+
+        Raises:
+            QiskitError: If the experiment does not have a backend set.
+        """
+        if self.backend is None:
```

Comment:
Since `center_frequency` is only needed to adjust the frequencies in the schedule for the shifts we can make this private. See here: a2bc3da . On second thoughts, from a user perspective it is useful to have a center frequency method since we can run in a relative mode. I therefore renamed the existing `_center_frequency` to `_backend_center_frequency` and introduced the `center_frequency` property which behaves differently. See here: 8bdce88